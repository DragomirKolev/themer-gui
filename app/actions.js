// Actions

export const COLOR_CHANGE = 'COLOR_CHANGE';
export const FOCUS_MODE_TOGGLE = 'FOCUS_MODE_TOGGLE ';
export const EXPORT_DIALOG_OPEN = 'EXPORT_DIALOG_OPEN';
export const HELP_DIALOG_OPEN = 'HELP_DIALOG_OPEN';
export const CLOSE_DIALOGS = 'CLOSE_DIALOGS';
export const SET_EXPORT_OPTION = 'SET_EXPORT_OPTION';
export const EXPORT_THEMES = 'EXPORT_THEMES';

// Action generators

export const colorChange = (colorSetKey, colorKey, value) => ({
  type: COLOR_CHANGE,
  colorSetKey,
  colorKey,
  value,
});

export const focusModeToggle = (isLight) => ({
  type: FOCUS_MODE_TOGGLE,
  isLight,
});

export const exportDialogOpen = () => ({
  type: EXPORT_DIALOG_OPEN,
});

export const helpDialogOpen = () => ({
  type: HELP_DIALOG_OPEN,
});

export const closeDialogs = () => ({
  type: CLOSE_DIALOGS,
});

export const setExportOption = (option, value) => ({
  type: SET_EXPORT_OPTION,
  option,
  value,
});

export const exportThemes = () => ({
  type: EXPORT_THEMES,
});
