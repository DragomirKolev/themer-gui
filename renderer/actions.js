// Actions

export const COLOR_CHANGE = 'COLOR_CHANGE';
export const FOCUS_MODE_TOGGLE = 'FOCUS_MODE_TOGGLE ';
export const EXPORT_DIALOG_OPEN = 'EXPORT_DIALOG_OPEN';
export const TIPS_DIALOG_OPEN = 'TIPS_DIALOG_OPEN';
export const CLOSE_DIALOGS = 'CLOSE_DIALOGS';
export const SET_EXPORT_OPTION = 'SET_EXPORT_OPTION';
export const EXPORT_PROGRESS_DIALOG_OPEN = 'EXPORT_PROGRESS_DIALOG_OPEN';
export const EXPORT_CANCELLED = 'EXPORT_CANCELLED';
export const EXPORT_PROGRESS = 'EXPORT_PROGRESS';
export const EXPORT_ERROR = 'EXPORT_ERROR';
export const EXPORT_COMPLETE = 'EXPORT_COMPLETE';
export const EXPORT_RESET = 'EXPORT_RESET';
export const PREFILL_DIALOG_OPEN = 'PREFILL_DIALOG_OPEN';
export const PREFILL_COLOR_SET_SELECTION_CHANGE = 'PREFILL_COLOR_SET_SELECTION_CHANGE';
export const PREFILL_WITH_COLOR_SET = 'PREFILL_WITH_COLOR_SET';
export const SAVE_COMPLETE = 'SAVE_COMPLETE';
export const OPEN_COMPLETE = 'OPEN_COMPLETE';
export const RESET_STATE = 'RESET_STATE';
export const UPDATE_READY = 'UPDATE_READY';
export const NEXT_TIP = 'NEXT_TIP';
export const PREVIOUS_TIP = 'PREVIOUS_TIP';

// Action generators

export const colorChange = (colorSetKey, colorKey, value) => ({
  type: COLOR_CHANGE,
  colorSetKey,
  colorKey,
  value,
});

export const focusModeToggle = (isLight) => ({
  type: FOCUS_MODE_TOGGLE,
  isLight,
});

export const exportDialogOpen = () => ({
  type: EXPORT_DIALOG_OPEN,
});

export const tipsDialogOpen = () => ({
  type: TIPS_DIALOG_OPEN,
});

export const closeDialogs = () => ({
  type: CLOSE_DIALOGS,
});

export const setExportOption = (option, value) => ({
  type: SET_EXPORT_OPTION,
  option,
  value,
});

export const exportProgressDialogOpen = () => ({
  type: EXPORT_PROGRESS_DIALOG_OPEN,
});

export const exportCancelled = () => ({
  type: EXPORT_CANCELLED,
});

export const exportProgress = (status) => ({
  type: EXPORT_PROGRESS,
  status,
});

export const exportError = (message) => ({
  type: EXPORT_ERROR,
  message,
});

export const exportComplete = (exportedPath) => ({
  type: EXPORT_COMPLETE,
  exportedPath,
});

export const exportReset = () => ({
  type: EXPORT_RESET,
});

export const prefillDialogOpen = () => ({
  type: PREFILL_DIALOG_OPEN,
});

export const prefillColorSetSelectionChange = (selection) => ({
  type: PREFILL_COLOR_SET_SELECTION_CHANGE,
  selection,
});

export const prefillWithColorSet = (colorSet) => ({
  type: PREFILL_WITH_COLOR_SET,
  colorSet,
});

export const saveComplete = (filePath) => ({
  type: SAVE_COMPLETE,
  filePath,
});

export const openComplete = (fileData) => ({
  type: OPEN_COMPLETE,
  fileData,
});

export const resetState = () => ({
  type: RESET_STATE,
});

export const updateReady = () => ({
  type: UPDATE_READY,
});

export const nextTip = () => ({
  type: NEXT_TIP,
});

export const previousTip = () => ({
  type: PREVIOUS_TIP,
});
