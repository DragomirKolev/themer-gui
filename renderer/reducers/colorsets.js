import { COLOR_CHANGE, PREFILL_WITH_COLOR_SET } from '../actions';

const defaultColorSets = {
  dark: {
    shade0: '',
    shade1: '',
    shade2: '',
    shade3: '',
    shade4: '',
    shade5: '',
    shade6: '',
    shade7: '',
    accent0: '',
    accent1: '',
    accent2: '',
    accent3: '',
    accent4: '',
    accent5: '',
    accent6: '',
    accent7: '',
  },
  light: {
    shade0: '',
    shade1: '',
    shade2: '',
    shade3: '',
    shade4: '',
    shade5: '',
    shade6: '',
    shade7: '',
    accent0: '',
    accent1: '',
    accent2: '',
    accent3: '',
    accent4: '',
    accent5: '',
    accent6: '',
    accent7: '',
  },
};

export const colorSetsReducer = (state = defaultColorSets, action) => {
  switch (action.type) {
    case COLOR_CHANGE:
      const { colorSetKey, colorKey, value } = action;
      const ret = {
        ...state,
        [colorSetKey]: {
          ...state[colorSetKey],
          [colorKey]: value,
        },
      };
      return ret;
    case PREFILL_WITH_COLOR_SET:
      return {
        dark: {
          ...defaultColorSets.dark,
          ...action.colorSet.dark,
        },
        light: {
          ...defaultColorSets.light,
          ...action.colorSet.light,
        },
      };
    default:
      return state;
  }
};
