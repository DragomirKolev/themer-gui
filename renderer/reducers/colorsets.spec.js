import { colorSetsReducer } from './colorsets';
import { colorChange, prefillWithColorSet, distributeShades } from '../actions';

describe('colorSetsReducer', () => {
  it('should initialize with all required keys and empty values', () => {
    expect(colorSetsReducer(undefined, {})).toMatchSnapshot();
  });
  it('should properly update colors', () => {
    const state0 = colorSetsReducer(undefined, {});
    const state1 = colorSetsReducer(state0, colorChange('dark', 'shade2', '#555'));
    const state2 = colorSetsReducer(state1, colorChange('light', 'accent0', 'red'));
    expect(state2).toMatchSnapshot();
  });
  it('should support setting the color set all at once', () => {
    const initialState = colorSetsReducer(undefined, colorChange('light', 'accent0', 'red'));
    const prefilledState = colorSetsReducer(initialState, prefillWithColorSet({ dark: { shade0: '#222' }}));
    expect(prefilledState).toMatchSnapshot();
  });
  it('should support distributing shades', () => {
    const noOpState = colorSetsReducer(undefined, distributeShades());
    expect(noOpState).toMatchSnapshot();
    const distributedState = colorSetsReducer({
      dark: {
        shade0: '#000000',
        shade1: '',
        shade2: '',
        shade3: '',
        shade4: '',
        shade5: '',
        shade6: '',
        shade7: '#ffffff',
        accent0: '',
        accent1: '',
        accent2: '',
        accent3: '',
        accent4: '',
        accent5: '',
        accent6: '',
        accent7: '',
      },
      light: {
        shade0: '#ffffff',
        shade1: '',
        shade2: '',
        shade3: '',
        shade4: '',
        shade5: '',
        shade6: '',
        shade7: '#000000',
        accent0: '',
        accent1: '',
        accent2: '',
        accent3: '',
        accent4: '',
        accent5: '',
        accent6: '',
        accent7: '',
      },
    }, distributeShades());
    expect(distributedState).toMatchSnapshot();
  });
});
