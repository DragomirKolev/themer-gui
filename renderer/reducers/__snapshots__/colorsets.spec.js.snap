// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorSetsReducer should initialize with all required keys and empty values 1`] = `
Object {
  "dark": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "",
    "shade1": "",
    "shade2": "",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
  "light": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "",
    "shade1": "",
    "shade2": "",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
}
`;

exports[`colorSetsReducer should properly update colors 1`] = `
Object {
  "dark": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "",
    "shade1": "",
    "shade2": "#555",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
  "light": Object {
    "accent0": "red",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "",
    "shade1": "",
    "shade2": "",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
}
`;

exports[`colorSetsReducer should support distributing shades 1`] = `
Object {
  "dark": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "",
    "shade1": "",
    "shade2": "",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
  "light": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "",
    "shade1": "",
    "shade2": "",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
}
`;

exports[`colorSetsReducer should support distributing shades 2`] = `
Object {
  "dark": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "#000000",
    "shade1": "#242424",
    "shade2": "#494949",
    "shade3": "#6D6D6D",
    "shade4": "#929292",
    "shade5": "#B6B6B6",
    "shade6": "#DBDBDB",
    "shade7": "#FFFFFF",
  },
  "light": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "#FFFFFF",
    "shade1": "#DBDBDB",
    "shade2": "#B6B6B6",
    "shade3": "#929292",
    "shade4": "#6D6D6D",
    "shade5": "#494949",
    "shade6": "#242424",
    "shade7": "#000000",
  },
}
`;

exports[`colorSetsReducer should support setting the color set all at once 1`] = `
Object {
  "dark": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "#222",
    "shade1": "",
    "shade2": "",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
  "light": Object {
    "accent0": "",
    "accent1": "",
    "accent2": "",
    "accent3": "",
    "accent4": "",
    "accent5": "",
    "accent6": "",
    "accent7": "",
    "shade0": "",
    "shade1": "",
    "shade2": "",
    "shade3": "",
    "shade4": "",
    "shade5": "",
    "shade6": "",
    "shade7": "",
  },
}
`;
